lib/extensions/conditional/active_record/base.rb

```diff
@@ -62,7 +62,11 @@ def dependent_class
       raise NotImplementedError
     end
 
-    def resolve_conditional_for(_course_user)
+    # Resolve the conditionals for the given course user
+    #
+    # @param [CourseUser] course_user The user with conditionals to be resolved
```

Comment:
Should this document the fact that the job is returned, and that people can wait for the job to finish?